src/main/java/com/ecommerce/ECommerceApp/model/Order.java

```diff
@@ -25,7 +26,7 @@ public Order() {}
      * @param userID the user ID associated with the order
      * @param totalAmount the total amount of the order
      */
-    public Order(int orderID, LocalDateTime orderDate, int userID, float totalAmount) {
+    public Order(int orderID, LocalDateTime orderDate, int userID, BigDecimal totalAmount) {
```

Comment:
Replaced float with BigDecimal for monetary values to prevent rounding issues in financial calculations, ensuring precision and reliability. As Martin Fowler discusses in ["Technical Debt"](https://martinfowler.com/bliki/TechnicalDebt.html), addressing data accuracy early on reduces technical debt and avoids potential issues down the line.